File: 1460-number-of-substrings-containing-all-three-characters/number-of-substrings-containing-all-three-characters.py
class Solution:
    def numberOfSubstrings(self, s: str) -> int:
        count=0
        n=len(s)
        last={
            'a':-1,
            'b':-1,
            'c':-1
        }
        for i in range(n):
            last[s[i]]=i
            minimum=min(last['a'],last['b'],last['c'])
            if minimum!=-1:
                count+=minimum+1
        return count

File: 1460-number-of-substrings-containing-all-three-characters/number-of-substrings-containing-all-three-characters.js
/**
 * @param {string} s
 * @return {number}
 */
var numberOfSubstrings = function(s) {
    let last={
        a:-1,
        b:-1,
        c:-1
    };
    let count=0;
    for(let i=0;i<s.length;i++){
        last[s[i]]=i;
        let minimum=Math.min(last.a,last.b,last.c)
        if(minimum!==-1){
            count+=minimum+1;
        }
    }
    return count;
};